packages/components/src/FormField/FormField.test.tsx

```diff
@@ -270,6 +277,32 @@ describe("FormField", () => {
       });
     });
 
+    describe("when nested/structured", () => {
+      it("should display errors", async () => {
+        const { getByText } = render(
+          <Form onSubmit={jest.fn()}>
+            <FormField
+              name="parent.0.child"
+              placeholder="foo"
+              validations={{
+                required: {
+                  value: true,
+                  message: "field foo is required",
+                },
+              }}
+            />
+            <button type="submit">Submit</button>,
+          </Form>,
+        );
+
+        await act(async () => {
```

Comment:
essentially, yes. 

I could also do a `await findByText('field foo is required')`. maybe that would be preferable? if you think so, can you articulate _why_ we think that would be preferable?

Reply:
I'd consider putting the _fireEvent_ outside the _act_ and after it doing something like:

```ts
fireEvent(..)

// flush state changes caused by react-hook-form
await act(async () => {})
```

To make it more obvious

Reply:
Hrmm. I think I disagree. We have 3 options...

1. async act

```
await act(async () => fireEvent.click(...));
expect(getByText(...)).toBeInTheDocument();
```

2. findBy

```
fireEvent.click(...);
expect(await findByText(...)).toBeInTheDocument();
```

3. separate async act

```
fireEvent.click(...);
await act(async () => {});
expect(getByText(...)).toBeInTheDocument();
```

- I don't feel that we benefit from separating 'async act' into its own statement
- I can see your point about making it more obvious, but I think developers will need to be more aware of both `findBy` and `async act` going forward.
- I don't think we'd want that (separate statement for `async act`) to be the pattern for everyone working with Atlantis form
- In the future when we switch to `await userEvent.click(...)`, that _should_ be a drop-in replacement for `await act(async () => fireEvent.click(...));`
- Happy to discuss more, perhaps with @ericchernuka because I know he has an opinion too... 😅

Reply:
I think its #1 or #2. #1 essentially mimics what `await userEvent.click` would do in my head. #2 essentially mimics how it would play out "User clicks, time passes and we wait for the message to show up" which uses the find by for th waiting.

Reply:
#2 lgtm as well. #1 would fail linting in the Jobber online repo which is why I'm flagging it

Reply:
Not that linting rules are always applicable but this is the rule that recommends against it: https://github.com/testing-library/eslint-plugin-testing-library/blob/main/docs/rules/no-unnecessary-act.md

Reply:
If you can go with no. 2 I would do that. I think no. 1 makes it difficult to grok for the next developer to understand the reason behind putting _fireEvent_ inside _act_. If the reason is to deal with some rerenders due to state changes that are not awaitable then being a bit explicit about is okay I think. _floating-ui_ also recommends this approach in their [docs](https://floating-ui.com/docs/react#testing)

Reply:
Also, I wouldn't block the PR on this

Reply:
I think I agree with you @rjdestigter. As a general pattern, I think it's easier for devs to think "most of the time I need to reach for findBy, and in the rare case that doesn't work, I can try an async act".

I've updated the test to use #2.